python/ray/train/trainer.py

```diff
@@ -40,13 +38,28 @@ def noop():
 
 logger = logging.getLogger(__name__)
 
-BACKEND_NAME_TO_CONFIG_CLS = {
-    "horovod": HorovodConfig,
-    "tensorflow": TensorflowConfig,
-    "torch": TorchConfig
+BACKEND_NAME_TO_CONFIG_CLS_NAME = {
+    "horovod": "HorovodConfig",
+    "tensorflow": "TensorflowConfig",
+    "torch": "TorchConfig"
 }
 
 
+# Import backend configurations dynamically since not all subdependencies
+# may be installed.
+def get_backend_config_cls(backend_name) -> type:
+    try:
+        import importlib
+        config_cls = getattr(
+            importlib.import_module(f"ray.train"
+                                    f".{backend_name}"),
+            BACKEND_NAME_TO_CONFIG_CLS_NAME[backend_name])
+        return config_cls
+    except ImportError:
+        raise ValueError(f"{backend_name} is not installed. Please "
+                         f"install {backend_name} to use this backend.")
```

Comment:
This handles the case if the user passes a `backend_name` string pretty cleanly. Do we need similar wrapping logic if the user directly passes in a `BackendConfig`? Will the error message in that scenario be clear?

Reply:
Good point, fixed!